TowerForge/TowerForge/LevelManager/Components/TFComponent.swift

```diff
@@ -7,8 +7,9 @@
 
 import Foundation
 
-class TFComponent: Identifiable {
-    var id = UUID()
+class TFComponent: Identifiable, Hashable {
+
+    var componentType: Enums.Components = .Default
```

Comment:
Since we changed the TFComponents inside the TFEntity to a Set from a dictionary, we need some way to make TFComponents hashable and equatable. The equivalence criteria I have chosen is their component type which I've added with an enum, as .Type does not conform to hashable. The default enum refers to TFComponent itself and it is necessary because TFComponent is a concrete class. I think it should be a protocol though since it wouldn't be instantiated directly anyway, then the placeholder .Default can be removed as well.

Reply:
hmm I see, i am not too sure about this since i have seen a dictionary implementation instead

Reply:
I'm not sure I understand what you mean, because I thought this was what we had discussed yesterday with regard to changing the dictionary to a Set since only one component of a given type will exist for an entity? Not an issue to revert back to dictionary though, I just thought having a set would ensure that we have an implicit CheckRep because having a maximum of one component of a given type at all times is the representation invariant right?

Reply:
Okay yup can i trust